src/intrinsics/ecma262/Date.js

```diff
@@ -189,7 +189,13 @@ export default function (realm: Realm): NativeFunctionValue {
 
   // ECMA262 20.3.3.2
   func.defineNativeMethod("parse", 1, (context, [string]) => {
-    throw new Error("TODO: Implement Date.parse");
+    let concreteString = string.throwIfNotConcrete();
+    let s = ToString(realm, concreteString);
+    if ((s.length > 0 && s.charAt(s.length - 1) !== 'Z') ||
```

Comment:
re partial mode check: Herman says that the trailing 'Z' check is only really needed in partial mode. Partial mode flag being a proxy for the desire to have a deterministic execution.

re strictlyMonotonicDateNow: No need for a test as it's only needed for testing. Just make sure that the value flows through properly.